backend/de.metas.printing/de.metas.printing.base/src/main/java/de/metas/printing/async/spi/impl/InvoicePDFConcatenateWorkpackageProcessor.java

```diff
@@ -0,0 +1,111 @@
+package de.metas.printing.async.spi.impl;
+
+import com.lowagie.text.Document;
+import com.lowagie.text.DocumentException;
+import com.lowagie.text.pdf.PdfCopy;
+import com.lowagie.text.pdf.PdfReader;
+import de.metas.async.api.IQueueDAO;
+import de.metas.async.model.I_C_Async_Batch;
+import de.metas.async.model.I_C_Queue_WorkPackage;
+import de.metas.async.spi.IWorkpackageProcessor;
+import de.metas.attachments.AttachmentEntryService;
+import de.metas.logging.TableRecordMDC;
+import de.metas.printing.api.IPrintingDAO;
+import de.metas.printing.model.I_C_Printing_Queue;
+import de.metas.util.Check;
+import de.metas.util.Services;
+import org.adempiere.ad.trx.api.ITrx;
+import org.adempiere.archive.api.IArchiveBL;
+import org.adempiere.service.ISysConfigBL;
+import org.compiere.SpringContextHolder;
+import org.compiere.model.I_AD_Archive;
+import org.slf4j.MDC;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * This processor process the enqueued <code>I_C_Printing_Queue</code>
+ *
+ * <ul> for each C_Printing_Queue selection</ul>
+ * <ul> fetches the PDF files</ul>
+ * <ul> and concatenates all pdf in one big pdf file per selection</ul>
+ *
+ * @author cg
+ */
+public class InvoicePDFConcatenateWorkpackageProcessor implements IWorkpackageProcessor
```

Comment:
aparently this processor is *not* about invoices, but about concatenating any printing items' archives' PDFs. please change the name accordingly